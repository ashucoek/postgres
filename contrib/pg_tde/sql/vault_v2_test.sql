\set tde_am tde_heap
\i sql/vault_v2_test.inc