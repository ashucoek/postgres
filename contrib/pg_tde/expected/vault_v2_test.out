CREATE EXTENSION IF NOT EXISTS pg_tde;
\getenv root_token ROOT_TOKEN
SELECT pg_tde_add_database_key_provider_vault_v2('vault-incorrect',:'root_token','http://127.0.0.1:8200','DUMMY-TOKEN',NULL);
 pg_tde_add_database_key_provider_vault_v2 
-------------------------------------------
                                         1
(1 row)

-- FAILS
SELECT pg_tde_set_key_using_database_key_provider('vault-v2-key','vault-incorrect');
ERROR:  Invalid HTTP response from keyring provider "vault-incorrect": 404
CREATE TABLE test_enc(
	  id SERIAL,
	  k INTEGER DEFAULT '0' NOT NULL,
	  PRIMARY KEY (id)
	) USING tde_heap;
ERROR:  principal key not configured
HINT:  create one using pg_tde_set_key before using encrypted tables
SELECT pg_tde_add_database_key_provider_vault_v2('vault-v2',:'root_token','http://127.0.0.1:8200','secret',NULL);
 pg_tde_add_database_key_provider_vault_v2 
-------------------------------------------
                                         2
(1 row)

SELECT pg_tde_set_key_using_database_key_provider('vault-v2-key','vault-v2');
 pg_tde_set_key_using_database_key_provider 
--------------------------------------------
 
(1 row)

CREATE TABLE test_enc(
	  id SERIAL,
	  k INTEGER DEFAULT '0' NOT NULL,
	  PRIMARY KEY (id)
	) USING tde_heap;
INSERT INTO test_enc (k) VALUES (1);
INSERT INTO test_enc (k) VALUES (2);
INSERT INTO test_enc (k) VALUES (3);
SELECT * from test_enc;
 id | k 
----+---
  1 | 1
  2 | 2
  3 | 3
(3 rows)

SELECT pg_tde_verify_key();
 pg_tde_verify_key 
-------------------
 
(1 row)

DROP TABLE test_enc;
DROP EXTENSION pg_tde;
